syntax = "proto3";

package aggregation_server;

service AggregationServer {
  rpc Configure (ConfigureRequest) returns (ConfigureResponse);
  rpc Aggregate (AggregateRequest) returns (AggregateResponse);
  rpc Cleanup (CleanupRequest) returns (CleanupResponse);
}

message ConfigureRequest {
  string request_id = 1;
  int32 num_of_workers = 2;
}

message ConfigureResponse {
  string status = 1;
}

message AggregateRequest {
  string request_id = 1;
  string aggregation_type = 2;
  repeated float vectors = 3;
}

message AggregateResponse {
  repeated float result = 1;
}

message CleanupRequest {
  string request_id = 1;
}

message CleanupResponse {
  string status = 1;
}
